### PR TITLE
Implementation of PROPORTIONAL_TO_GENERATION_P balancetype

### PR DESCRIPTION
**Pull request by obrix**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

Implementation of PROPORTIONAL_TO_GENERATION_P balancetype : 

It reuses GenerationActionPowerDistributionStep class as implementation is the same, only the participation factor is computed differently.

Unsupported balance type test has been replaced by a small test of proportional to generation P balance type.

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

### CODE REVIEW

**Comment by obrix** [comment]
@geofjamg Can you confirm GenerationActionPowerDistributionStep should be renamed GenerationActivePowerDistributionStep ?

File: src/main/java/com/powsybl/openloadflow/network/util/GenerationActionPowerDistributionStep.java [hunk 1]
```diff
@@ -92,4 +105,19 @@ public double run(List<ParticipatingElement> participatingElements, int iteratio
...
     }
+
+    private double getParticipationFactor(LfGenerator generator) {
+        double factor;
+        switch (participationType) {
+            case MAX:
+                factor = generator.getMaxP() / generator.getDroop();
+                break;
+            case TARGET:
+                factor = generator.getTargetP();
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/util/GenerationActionPowerDistributionStep.java:116]: Finally, I think that the absolute value is required here.

**Comment by obrix** [comment]
I made the choice of removing getParticipationFactor from LfGenerator and to compute the participationFactor from GenerationActionPowerDistributionStep as it is dependant from the BalanceType configuration. 

This keeps the lfGenerator independant from GenerationActionPowerDistributionStep configuration but made mandatory the implementation of LfGenerator::getDroop.

@geofjamg Any opinion about that ? I think keeping GenerationActionPowerDistributionStep generic and independant is a good idea but I'm not sure about the relevance of exposing the droop parameter.

File: src/main/java/com/powsybl/openloadflow/network/util/GenerationActionPowerDistributionStep.java (hunk 1 above)
Comment (obrix) [reply src/main/java/com/powsybl/openloadflow/network/util/GenerationActionPowerDistributionStep.java:116]: Okay thanks I will modify the computation.

Comment (obrix) [reply src/main/java/com/powsybl/openloadflow/network/util/GenerationActionPowerDistributionStep.java:116]: Can you confirm it's only for the factor in targetP mode ?

**Comment by geofjamg** [comment]
> I made the choice of removing getParticipationFactor from LfGenerator and to compute the participationFactor from GenerationActionPowerDistributionStep as it is dependant from the BalanceType configuration.
> 
> This keeps the lfGenerator class independant from GenerationActionPowerDistributionStep configuration but made mandatory the implementation of LfGenerator::getDroop.
> 
> @geofjamg Any opinion about that ? I think keeping GenerationActionPowerDistributionStep generic and independant is a good idea but I'm not sure about the relevance of exposing the droop parameter.

Ok, good point.

**Review by geofjamg — APPROVED** [review]

**Pull request merged**